### ISSUE TITLE
🐞 Firestore error in Android when network is down

### ISSUE THREAD

**Issue by erohin** [invertase/react-native-firebase#2681]
## Issue

After `firebase.firestore().collection('collection').doc('app').get()` only in Android only when network is down
```
FATAL EXCEPTION: main
Process: com.sample.app, PID: 6891
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Throwable.getMessage()' on a null object reference
at io.invertase.firebase.firestore.ReactNativeFirebaseFirestoreCommon.rejectPromiseFirestoreException(ReactNativeFirebaseFirestoreCommon.java:33)
at io.invertase.firebase.firestore.ReactNativeFirebaseFirestoreDocumentModule.lambda$documentGet$2(ReactNativeFirebaseFirestoreDocumentModule.java:139)
at io.invertase.firebase.firestore.-$$Lambda$ReactNativeFirebaseFirestoreDocumentModule$u-MMi3PSmZNyC4Qqk9NTkmXWp8o.onComplete(Unknown Source:2)
at com.google.android.gms.tasks.zzj.run(Unknown Source:4)
at android.os.Handler.handleCallback(Handler.java:790)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6494)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```
From logs before this error:
```
(20.2.0) [OnlineStateTracker]: Could not reach Cloud Firestore backend. Backend didn't respond within 10 seconds

This typically indicates that your device does not have a healthy Internet connection at the moment. The client will operate in offline mode until it is able to successfully connect to the backend.
```
---

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [x] my application is an AndroidX application?
- [x] I am using `android/gradle.properties` `android.enableJetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy

```

#### `android/app/build.gradle`:

```groovy
```

#### `android/settings.gradle`:

```groovy
```

#### `MainApplication.java`:

```java
```

#### `AndroidManifest.xml`:

```xml
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
System:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.1 => 0.61.1
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [x] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `6.0.0`
- **`Firebase` module(s) you're using that has the issue:**
  - `Firestore`
- **Are you using `TypeScript`?**
  - `Y`
  
</p>
</details>

---

Think `react-native-firebase` is great? Please consider supporting all of the project maintainers and contributors by donating via our [Open Collective](https://opencollective.com/react-native-firebase/donate) where all contributors can submit expenses. [[Learn More]](https://invertase.io/oss/react-native-firebase/contributing/donations-expenses)

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by Ehesp** [comment]
Looks like is us not handling an exception when offline, will file as a bug. Thanks.

**Comment by Salakar** [comment]
Hey 👋 this issue should now be fixed in v6.0.1, please try it out - thanks. [[release notes]](https://invertase.io/oss/react-native-firebase/releases/v6.0.1)
